app/src/main/java/pm/gnosis/heimdall/ui/settings/general/GetInTouchActivity.kt

```diff
@@ -0,0 +1,78 @@
+package pm.gnosis.heimdall.ui.settings.general
+
+import android.content.ActivityNotFoundException
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import kotlinx.android.synthetic.main.screen_get_in_touch.*
+import pm.gnosis.heimdall.R
+import pm.gnosis.heimdall.reporting.ScreenId
+import pm.gnosis.heimdall.ui.base.BaseActivity
+import com.google.android.material.snackbar.Snackbar
+import pm.gnosis.svalinn.common.utils.snackbar
+
+
+class GetInTouchActivity : BaseActivity() {
+
+    override fun screenId() = ScreenId.GET_IN_TOUCH
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.screen_get_in_touch)
+
+        touch_toolbar_back_arrow.setOnClickListener {
+            onBackPressed()
+        }
+
+        touch_telegram.setOnClickListener {
+            openTelegramChannel()
+        }
+        touch_email.setOnClickListener {
+            sendEmail()
+        }
+        touch_gitter.setOnClickListener {
+            openGitterChannel()
+        }
+    }
+
+    private fun openTelegramChannel() {
+
+        var intent = try {
+            try {
+                //Check for Telegram Messenger App
+                packageManager.getPackageInfo("org.telegram.messenger", 0)
+            } catch (e: Exception) {
+                //Check for Telegram X App
+                packageManager.getPackageInfo("org.thunderdog.challegram", 0)
+            }
+            Intent(Intent.ACTION_VIEW, Uri.parse("tg://resolve?domain=${getString(R.string.gnosis_telegram_id)}"))
```

Comment:
I would actually just try to start this url and if no app can handle it, it will throw an error anyways.

Also can't the telegram app handle the `www.telegram.me` deeplink?

Reply:
it can, but android will show app chooser for the first time (choosing between browser and telegram)